### PR TITLE
Fix/git branch field add repo/4297

### PR DESCRIPTION
**Pull request by Sean-McQ**
It doesn't look like this field is _actually_ required by the catalog.cattle.io.clusterrepo model and the save method on the form would produce an error on save if this were the case. Just removing the required prop from the field for now.

Relates to https://github.com/rancher/dashboard/issues/4297

A future update will include enforceable form-field validation separate from the model validation already in place so we would be able to easily add this requirement back in on the form if needed when that feature is complete.

### CODE REVIEW

**Comment by vincent99** [comment]
_A_ branch is required, but it does default to `master` in the backend if not specified.  `main` is now the default for new GitHub repos, so it would probably be more helpful to set the value to `master` when you load the form if empty and keep calling it required.

**Comment by Sean-McQ** [comment]
Setting a default would probably be useful but it doesn't really resolve the underlying issue. If the field is required then the repo shouldn't be saved. Ideally the API would return a 400 error telling us we made a bad request with a decent error message but since we're pre-validating we'd need to return an error and then not send the request.

I'll see if I can't add a custom validator on the model if isGit returns true which seems to be the way we handle most other issues and potentially disable the save button if the model returns errors.

There's a more complicated use-case here with regards to switching between git and helm mode and requiring one or the other but that'll require a much more creative user error than just forgetting a branch name.

**Comment by vincent99** [comment]
A value _is_ ultimately required.  There has to be some branch name to pull when it is time to clone the repo.  The server has an implied default (of `master`), but does not report that in the schema, so we currently claim it's required, but don't don't tell you what the default is or make you provide a value.

So I would like to make all that explicitly visible to the user instead invisible magic.   Treat it as a field that is required in the UI, even though you won't literally get an error back from the API if not provided:
1. Leave the red `*`.
2. When bringing up the form, set the branch to `master` if the branch is empty (for create or edit).
3. On validation/save click, show message & prevent post/put unless there is a value provided.

File: models/catalog.cattle.io.clusterrepo.js [hunk 1]
```diff
@@ -29,6 +29,21 @@ export default class ClusterRepo extends SteveModel {
     return out;
   }
 
+  get customValidationRules() {
+    const out = [];
+
+    if (this.isGit) {
```
Comment (mantis-toboggan-md) [inline models/catalog.cattle.io.clusterrepo.js:35]: `this.isGit` is only true if `value.spec.gitRepo` is non-empty, so we're still able to make repos with neither a git repo url nor git branch specified despite both being marked as required. I think we need a bit fancier logic around disabling the create button on this page, where either `spec.url` or `spec.gitRepo` and `spec.gitBranch` must be defined.

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]

File: models/catalog.cattle.io.clusterrepo.js (hunk 1 above)
Comment (Sean-McQ) [reply models/catalog.cattle.io.clusterrepo.js:35]: Agreed 100% although the backend would kick back an error if you sent it a payload with neither url or gitRepo defined. You can also define both in the current UI without being able to see both fields at the same time to know what the problem is. It's probably outside of the scope of this ticket to completely rewrite and redesign the form though.

Pulling in @lvuch to define what we'd expect from the UX in this particular case.

**Comment by Sean-McQ** [comment]
@vincent99 I'm still not super comfortable defining a field as required if it's absence doesn't produce some sort of feedback in the user interface at some point.

Third version here removes the asterix from the field but provides text informing the user that "Master" will be defaulted to if no branch is specified.

image

**Review by mantis-toboggan-md — APPROVED** [review]

**Review by lvuch — APPROVED** [review]
lgtm

**Pull request merged**